src/main.rs

```diff
@@ -230,20 +231,70 @@ fn setup_logging(num_verbose: u64) {
         },
     }
 
-    env_logger::init();
+    // Configure the OpenTelemetry tracer, if requested.
+    //
+    // To enable the tracing functionality, set OTEL_EXPORTER_JAEGER_AGENT_HOST
+    // env to some suitable value (see below).
+    //
+    // The Jaeger layer emits traces under the service name of "iox" if not
+    // overwrote by the user by setting the env below. All configuration is
+    // sourced from the environment:
+    //
+    // - OTEL_SERVICE_NAME: emitter service name (iox by default)
+    // - OTEL_EXPORTER_JAEGER_AGENT_HOST: hostname/address of the collector
+    // - OTEL_EXPORTER_JAEGER_AGENT_PORT: listening port of the collector
+    //
+    let (opentelemetry, drop_handle) = match std::env::var("OTEL_EXPORTER_JAEGER_AGENT_HOST") {
+        Ok(_) => {
+            // Initialise the jaeger event emitter
+            let (tracer, drop_handle) = opentelemetry_jaeger::new_pipeline()
+                .with_service_name("iox")
+                .from_env()
+                .install()
+                .unwrap();
+
+            // Initialise the opentelemetry tracing layer, giving it the jaeger emitter
+            let opentelemetry = tracing_opentelemetry::layer().with_tracer(tracer);
+
+            (Some(opentelemetry), Some(drop_handle))
+        }
+        Err(_) => (None, None),
+    };
```

Comment:
Stylistic suggestion here. You can reduce the indentation level by one by using `.is_ok()`:

```suggestion
    let (opentelemetry, drop_handle) = if std::env::var("OTEL_EXPORTER_JAEGER_AGENT_HOST").is_ok() {
        // Initialise the jaeger event emitter
        let (tracer, drop_handle) = opentelemetry_jaeger::new_pipeline()
            .with_service_name("iox")
            .from_env()
            .install()
            .unwrap();

        // Initialise the opentelemetry tracing layer, giving it the jaeger emitter
        let opentelemetry = tracing_opentelemetry::layer().with_tracer(tracer);

        (Some(opentelemetry), Some(drop_handle))
    } else {
        (None, None)
    };
```

Reply:
Ha I actually wrote it with `is_ok()` first and then swapped it to match for some reason...

Looking at it `is_ok()` is definitely the way to go - neither arm uses the value in the result - I'll push a commit 👍

Reply:
yeah that was what made me stop and take a minute to look at it. Variants not being used seems like the wrong pattern generally :)